opencti-platform/opencti-graphql/src/modules/entitySetting/entitySetting-attributeUtils.ts

```diff
@@ -123,5 +124,8 @@ export const getMandatoryAttributesForSetting = async (
   entitySetting: BasicStoreEntityEntitySetting
 ) => {
   const attributes = await getEntitySettingSchemaAttributes(context, user, entitySetting);
+  if (isUserHasCapability(user, KNOWLEDGE_KNUPDATE_KNBYPASSFIELDS)) {
```

Comment:
Might be missing something but I don't think this is a good way of handling this as you will "lose" the fact that the attribute is mandatory.

I mean, having the capability the bypass them is interesting, having no clue they indeed should be mandatory is another.

Reply:
Currently, we simply return a list of attribute names: `[‘name’, ‘description’ ]`. And in the Front end, we try to match these names with the fields in the form.
You can return all the mandatory attributes and filter in the front-end. We didn't opt for this solution when we were brainstorming, because in front-end, you can always do what you want and work around it manually. And if you don't use the front end to create the entity, the validation has to be in the back end.

Reply:
I have same point of view as @Kedae here. 

We have an API `EntitySetting.mandatoryAttributes` in graphql that is returning which are the required attributes for an entity type. No matter you have the bypass capability or not, those fields stay required. This solution makes the result of this API inconsistent depending of your capabilities and could lead to misunderstanding or issues depending on how the community is using this API.

Moreover, this `getMandatoryAttributesForSetting` function is also used in the middleware to prevent deleting a relationship attached to required stuff. I have the feeling that could lead to issues also here?

---

> in front-end, you can always do what you want and work around it manually

Even with this solution you can still do what you want and inspect code to remove front validation.

> And if you don't use the front end to create the entity, the validation has to be in the back end.

Don't you already done it in `schema-validator.ts`? Because this `getMandatoryAttributesForSetting` function is not about validating inputs so I don't get it. For what I understand you have already done the check on configured required attributes in `schema-validator.ts` so you cannot fake the API no matter what, we good on this point. 

It's just the way to handle those required attributes in UI that, in my opinion, need to be done in frontend based on capabilities and not on the result of this `EntitySetting.mandatoryAttributes` API.

Reply:
I can't find any cases where the ‘deleteRelationsByFromAndTo’ method is triggered with customisable mandatory attributes.

Reply:
Acceptable solution. Will be changed later if necessary